### DISCUSSION TITLE
Detect workspace change

### DISCUSSION THREAD

**Discussion by rzarai** [i3/i3#4634] [Q&A] [answered]
Hello everyone.  
Is it possible to detect the workspace change, and if so, can I for example write to file the name or number of the workspace I changed to?  
For example, I switch to workspace 1, then the "1" or "1: workspace one" is getting written in the text file. Is it possible to do so?

**Comment by orestisfl** [comment, marked as answer]
Yes, see the IPC interface: https://i3wm.org/docs/ipc.html

You can use `i3-msg -t subscribe -m '["workspace"]'` and parse the json output or a third party library, e.g. https://github.com/altdesktop/i3ipc-python

**Reply by Airblader** [reply]
Just as an addendum, the information could also be obtained by spying on the root window's `_NET_CURRENT_DESKTOP` atom (`xprop -root -spy _NET_CURRENT_DESKTOP`). The value can then be looked up in `_NET_DESKTOP_NAMES` (again on the root window).

This is more cumbersome, but portable to all EWMH-compliant window managers.
